src/main/java/org/shanerx/tradeshop/utils/gsonprocessing/typeadapters/ConfigurationSerializableAdapter.java

```diff
@@ -25,33 +25,36 @@
 
 package org.shanerx.tradeshop.utils.gsonprocessing.typeadapters;
 
-import com.google.gson.JsonDeserializationContext;
-import com.google.gson.JsonDeserializer;
-import com.google.gson.JsonElement;
-import com.google.gson.JsonParseException;
-import com.google.gson.JsonSerializationContext;
-import com.google.gson.JsonSerializer;
+import com.google.gson.*;
 import com.google.gson.reflect.TypeToken;
 import org.bukkit.configuration.serialization.ConfigurationSerializable;
 import org.bukkit.configuration.serialization.ConfigurationSerialization;
+import org.shanerx.tradeshop.TradeShop;
 import org.shanerx.tradeshop.utils.debug.Debug;
 import org.shanerx.tradeshop.utils.debug.DebugLevels;
 
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
 import java.lang.reflect.Type;
 import java.util.LinkedHashMap;
 import java.util.Map;
 
 //Based off of ConfigurationSerializableAdapter by Schottky <https://www.spigotmc.org/members/schottky.632864/> @ https://www.spigotmc.org/threads/configurationserializable-to-json-using-gson.467776/
 public class ConfigurationSerializableAdapter implements JsonSerializer<ConfigurationSerializable>, JsonDeserializer<ConfigurationSerializable> {
 
+    private static final String LINEFEED_ESCAPE = "##**M7a5b9c0fe874g398ff**##";
```

Comment:
You could just add an extra \ to the '\n' so it would be '\\n'

Pretty sure that would fix the problem since the line feed would be passed on

Reply:
That would open more issues - what if someone literally has the string \n? Then we need to escape to \\n. Which means we have to escape every \ as well... it becomes complicated quickly